.github/workflows/publish_pypi.yml

```diff
@@ -0,0 +1,99 @@
+name: Publish Python 🐍 distribution 📦 to PyPI
+
+on:
+  push:
+    branches:
+      - main                              # Trigger on push to main
+      - pypi-instructions-for-publishing  # Add other branch names if needed
+  pull_request:
+    branches:
+      - main 
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package-path:
+          - packages/happtiq_commons_gen_ai
+          - packages/happtiq_commons_google_cloud
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
```

Comment:
@kaanHapptiq before you build you need to switch to the right directory first
`cd packages/happtiq_commons_gen_ai`